src/server/scripts/Kalimdor/HallsOfOrigination/halls_of_origination.cpp

```diff
@@ -0,0 +1,414 @@
+/*
+ * Copyright (C) 2008-2018 TrinityCore <https://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* To-do: */
+// - Summoner Enabler dummy npc: unknown purpose (at the entrance)
+// - AreaTrigger: unknown purpose (id: 2275; pos: hole in the left wall at the entrance)
+// - Find out what Dummy Nuke (68991) spell does.
+// - Spatial Flux won't enter combat on second aggro from creature group (but it should).
+
+#include "ScriptMgr.h"
+#include "CreatureGroups.h"
+#include "GameObject.h"
+#include "GridNotifiers.h"
+#include "InstanceScript.h"
+#include "PassiveAI.h"
+#include "ScriptedGossip.h"
+#include "SpellAuraEffects.h"
+#include "SpellScript.h"
+#include "TemporarySummon.h"
+#include "halls_of_origination.h"
+
+enum Spells
+{
+    SPELL_SUBMERGE                      = 76084,
+
+    // Isiset trash and adds
+    SPELL_ARCANE_ENERGY                 = 74881,
+    SPELL_ARCANE_BURST                  = 74888, // On retail not working! Should probably be cast at full energy.
+    SPELL_SPAWN_ENERGY_FLUX_TRASH       = 82382, // Makes random player cast Summon Energy Flux
+    SPELL_ENERGY_FLUX_BEAM_TRASH        = 82377, // Makes nearby Spatial Flux cast visual beam
+    SPELL_ENERGY_FLUX_PERIODIC          = 74044,
+};
+
+enum Events
+{
+    // Spatial Flux
+    EVENT_DUMMY_NUKE,
```

Comment:
Event enum starts on 0 instead of 1 here.
Does it work as intended?

Reply:
Dummy Nuke is unscripted, unknown purpose... I will still update it to 1 👍

Reply:
Saw your reply to the previous one just now.
Feel free to ignore my previous comment.